.travis.yml

```diff
@@ -4,8 +4,15 @@
 
 osx_image: xcode9.2
 language: objective-c
-cache: cocoapods
+cache:
+  - cocoapods
+  - bundler
+  - npm
 install:
+  - nvm install 8.9.4
+  - npm install
+  - npm run build
+  - (! git status --porcelain | grep -q .) || (echo "Please ensure you have ran `npm run gulp`to generate the templated classes." && false)
```

Comment:
Can we use the same script that @matteo-hertel created for the android codebase?